daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NodeInfo.scala

```diff
@@ -452,9 +454,30 @@ object NodeInfo extends Enum {
       }
     }
 
+    trait PrimThrowsIllegalArgument {
+      def fromString(s: String): DataValuePrimitive
+      def fromXMLString(s: String): DataValuePrimitive = {
+        try {
+          fromString(s)
+        } catch {
+          case iae: IllegalArgumentException =>
+            throw new InvalidPrimitiveDataException("%s is not a valid %s".format(iae.getMessage, this))
+        }
+      }
+    }
+
     trait PrimNumeric { self: Numeric.Kind =>
       def isValidRange(n: Number): Boolean
       def fromNumber(n: Number): DataValueNumber
+      def fromString(s: String): DataValueNumber
```

Comment:
Should  fromString be protected? It is intended to override. 
fromXMLString is more explicit about the nature of the string being converted. I.e., it came from XML; hence, strings like "&#x31;" should be equivalent to "1". I don't know that it is called in any context where the string argument isn't already converted though, such that there won't be character numeric entities or other goop to deal with.

Reply:
fromString will be protected now. As far as I can tell, I believe the string is already converted. I at least haven't run into any issues in any of our tests or any of the schema tests. I will add a test to verify this though.